references/cli/ls.go

```diff
@@ -33,98 +32,64 @@ func NewListCommand(c types.Args, ioStreams cmdutil.IOStreams) *cobra.Command {
 			if err != nil {
 				return err
 			}
-			newClient, err := client.New(c.Config, client.Options{Scheme: c.Schema})
+			newClient, err := c.GetClient()
 			if err != nil {
 				return err
 			}
-			appName, err := cmd.Flags().GetString(App)
+			namespace, err := cmd.Flags().GetString(Namespace)
 			if err != nil {
 				return err
 			}
-			printComponentList(ctx, newClient, appName, env, ioStreams)
-			return nil
+			if namespace == "" {
+				namespace = env.Namespace
+			}
+			return printApplicationList(ctx, newClient, namespace, ioStreams)
 		},
 		Annotations: map[string]string{
 			types.TagCommandType: types.TypeApp,
 		},
 	}
-	cmd.PersistentFlags().StringP(App, "", "", "specify the name of application")
+	cmd.PersistentFlags().StringP(Namespace, "n", "", "specify the namespace the application want to list, default is the current env namespace")
 	return cmd
 }
 
-func printComponentList(ctx context.Context, c client.Reader, appName string, env *types.EnvMeta,
-	ioStreams cmdutil.IOStreams) {
-	deployedComponentList, err := common.ListComponents(ctx, c, common.Option{
-		AppName:   appName,
-		Namespace: env.Namespace,
-	})
-	if err != nil {
-		ioStreams.Infof("listing services: %s\n", err)
-		return
-	}
-
-	fetcher := func(name string) (*v1alpha2.Application, error) {
-		var app = new(v1alpha2.Application)
-		err := c.Get(ctx, client.ObjectKey{Name: name, Namespace: env.Namespace}, app)
-		return app, err
-	}
-	all := mergeStagingComponents(deployedComponentList, env, ioStreams, fetcher)
+func printApplicationList(ctx context.Context, c client.Reader, namespace string, ioStreams cmdutil.IOStreams) error {
 	table := newUITable()
-	table.AddRow("SERVICE", "APP", "TYPE", "TRAITS", "STATUS", "CREATED-TIME")
-	for _, a := range all {
-		traitAlias := strings.Join(a.TraitNames, ",")
-		table.AddRow(a.Name, a.App, a.WorkloadName, traitAlias, a.Status, a.CreatedTime)
+	table.AddRow("APP", "COMPONENT", "TYPE", "TRAITS", "PHASE", "HEALTHY", "STATUS", "CREATED-TIME")
```

Comment:
```suggestion
	table.AddRow("APP", "COMPONENT", "TYPE", "TRAITS", "PHASE", "HEALTHY", "ERROR-MESSAGE", "CREATED-TIME")
```

I learnt from your implementation, `HEALTHY` stands whether the component is healthy, and `STATUS` is its error message. But if there are data listed, I think you can ignore this suggestion.